js/signup.js

```diff
@@ -49,18 +59,25 @@ pwChkInput.addEventListener("focusout", pwCheckValidCheck);
 // 중복 function 실행으로 인한 변수 추가 (email, pw, pwchk)
 // validOption - 정상일 경우 type = none, 오류일 경우 type = valid_error
 // return 을 이용하여 submit에서 정성/오류 체크 가능하도록 수정
-function emailValidCheck() {
+async function emailValidCheck() {
   const validOption = { type: "none", text: "" };
+  const id = emailInput.value;
 
-  if (emailInput.value === "") {
+  if (id === "") {
     validOption.type = "valid_error";
     validOption.text = "이메일을 입력해 주세요.";
-  } else if (!emailRegex.test(emailInput.value)) {
-    validOption.type = "valid_error";
-    validOption.text = "올바른 이메일 주소가 아닙니다.";
-  } else if (emailInput.value === "test@codeit.com") {
-    validOption.type = "valid_error";
-    validOption.text = "이미 사용 중인 이메일입니다.";
+  } else {
+    await postReqeustApi("api/check-email", {
+      email: id,
+    }).catch((error) => {
+      if (error.message === "400") {
+        validOption.type = "valid_error";
+        validOption.text = "올바른 이메일 주소가 아닙니다.";
+      } else if (error.message === "409") {
+        validOption.type = "valid_error";
+        validOption.text = "이미 사용 중인 이메일입니다.";
+      }
+    });
```

Comment:
지금과 같이 작성하셔도 동작하는데는 아무 문제가 없으나, async/await에서 에러처리는 try-catch 문과 주로 쓰입니다 .

```suggestion
try {
            await postReqeustApi("api/check-email", {
                email: id,
            });
        } catch (error) {
            if (error.message === "400") {
                validOption.type = "valid_error";
                validOption.text = "올바른 이메일 주소가 아닙니다.";
            } else if (error.message === "409") {
                validOption.type = "valid_error";
                validOption.text = "이미 사용 중인 이메일입니다.";
            }
        }
```

위와 같이 작성해주면 좀 더 보편적인 스타일이 됩니다.